tests/php/ORM/DataListEagerLoadingTest.php

```diff
@@ -1288,5 +1289,226 @@ public function testEagerLoadingSharedRelations()
                 $this->assertInstanceOf(EagerLoadedList::class, $eagerLoaded2);
             }
         }
+        $this->assertGreaterThan(1, $loop1Count);
+        $this->assertGreaterThan(1, $loop2Count);
+    }
+
+    public function testInvalidAssociativeArray(): void
+    {
+        $this->expectException(LogicException::class);
+        $this->expectExceptionMessage(
+            'Value of associative array must be a callable.'
+            . 'If you don\'t want to pre-filter the list, use an indexed array.'
+        );
+        EagerLoadObject::get()->eagerLoad(['HasManyEagerLoadObjects' => 'HasManyEagerLoadObjects']);
+    }
+
+    public function provideNoLimitEagerloadingQuery(): array
```

Comment:
Do we need to test/implement `belongs_many_many`?

Reply:
Done - all of the many_many relation types are built the same so was pretty easy to refactor for the extra types.